src/axom/quest/examples/CMakeLists.txt

```diff
@@ -215,63 +215,49 @@ endif()
 
 if (ENABLE_FORTRAN)
 
-    blt_list_append(TO quest_example_fortran_depends ELEMENTS mpi IF ENABLE_MPI)
+    set(quest_fortran_examples quest_signed_distance_interface)
+    
+    # The inout Fortran example fails to compile with hipcc/amdflang in debug configurations
+    if("${CMAKE_Fortran_COMPILER}" MATCHES "amdflang$" AND "${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
+        set(_has_inout_fortran_example FALSE)
+    else()
+        set(_has_inout_fortran_example TRUE)
+        blt_list_append(TO quest_fortran_examples ELEMENTS quest_inout_interface)
+    endif()
 
-    blt_add_executable(
-        NAME       quest_signed_distance_interface_F_ex
-        SOURCES    quest_signed_distance_interface.F
-        OUTPUT_DIR ${EXAMPLE_OUTPUT_DIRECTORY}
-        DEPENDS_ON ${quest_example_depends} ${quest_example_fortran_depends}
-        FOLDER     axom/quest/examples
-        )
+    blt_list_append(TO quest_example_fortran_depends ELEMENTS mpi IF ENABLE_MPI)
 
-    # This target fails to compile with hipcc/amdflang on debugs
-    if(NOT "${CMAKE_Fortran_COMPILER}" MATCHES "amdflang$" OR NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
+    foreach(_t ${quest_fortran_examples})
+        set(_example_name "${_t}_F_ex")
         blt_add_executable(
-            NAME        quest_inout_interface_F_ex
-            SOURCES     quest_inout_interface.F
-            OUTPUT_DIR  ${EXAMPLE_OUTPUT_DIRECTORY}
-            DEPENDS_ON  ${quest_example_depends} ${quest_example_fortran_depends}
-            FOLDER      axom/quest/examples
-            )
+            NAME       ${_example_name}
+            SOURCES    ${_t}.F
+            OUTPUT_DIR ${EXAMPLE_OUTPUT_DIRECTORY}
+            DEPENDS_ON ${quest_example_depends} ${quest_example_fortran_depends}
+            FOLDER     axom/quest/examples)
 
-        if (NOT ENABLE_CUDA)
-            if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "IntelLLVM")
-                set_target_properties(quest_inout_interface_F_ex
+        # When CUDA is enabled, BLT will determine the correct linker, so don't override it here
+        if(NOT ENABLE_CUDA)
+            if ("${CMAKE_Fortran_COMPILER_ID}" STREQUAL "Intel" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "IntelLLVM")
```

Comment:
It's difficult to see, but the main fix in this PR is changing the check from the `CXX` compiler to the `Fortran`

```diff
- if ("${CMAKE_CXX_COMPILER_ID}"     STREQUAL "Intel" ...
+ if ("${CMAKE_Fortran_COMPILER_ID}" STREQUAL "Intel" ...
```

Reply:
Thanks for specifically pointing this out, @kennyweiss .